src/main/java/software/amazon/nio/spi/s3/NotYetImplementedException.java

```diff
@@ -0,0 +1,7 @@
+package software.amazon.nio.spi.s3;
+
+public class NotYetImplementedException extends RuntimeException {
```

Comment:
Why not just use the Java `UnsupportedOperationException`?

Reply:
Good point. That was my original approach, but by doing so, `FileSystems.newFileSystem(uri, Collections.emptyMap())` fails with
> java.nio.file.ProviderNotFoundException: Provider "s3" not found

which is quite a _misleading_ message. When reading the message, it can make one doubt that the provider is **actually** present (and correctly working) or not :( 

![image](https://github.com/awslabs/aws-java-nio-spi-for-s3/assets/283778/68c05381-15c5-4f11-8b2a-025b56fa7472)